packages/cli/src/commands/source/push/add.ts

```diff
@@ -154,4 +172,61 @@ export default class SourcePushAdd extends Command {
     );
     throw err;
   }
+
+  private splitByChunkAndUpload(
```

Comment:
The main function of this PR: 

Create and return two function: 

* One to push documents (send) which accumulate documents until they reach a maxContentLength size. When the limit is reached, push the documents to Coveo (`uploadBatch` function).
* One to "close" when there's no more files to parse and convert to `DocumentBuilder`, so that we can flush the remaining queue (remaining documents in `accumulator.chunks`)

Reply:
Nicely done